### PR TITLE
Use ordinal string operations

### PR DESCRIPTION
**Pull request by Rob-Hague**
This PR adds an .editorconfig file enabling [CA1310](https://learn.microsoft.com/en-gb/dotnet/fundamentals/code-analysis/quality-rules/ca1310) (Specify StringComparison for correctness) at 'error' severity and fixes violations by using ordinal string operations.

Ref https://github.com/connamara/quickfixn/issues/713#issuecomment-1484128288

### CODE REVIEW

File: QuickFIXn/Message/Message.cs [hunk 1]
```diff
@@ -681,7 +681,7 @@ public void ReverseRoute(Header header)
                 this.Header.RemoveField(Tags.OnBehalfOfLocationID);
                 this.Header.RemoveField(Tags.DeliverToLocationID);
 
-                if (beginString.CompareTo("FIX.4.1") >= 0)
+                if (string.CompareOrdinal(beginString, "FIX.4.1") >= 0)
```
Comment (maxnis) [inline QuickFIXn/Message/Message.cs:684]: As recommended by Microsoft - it's better to use Equals method to test equality, isn't it ?

Comment (Rob-Hague) [reply QuickFIXn/Message/Message.cs:684]: You're right, but here we are not checking for equality (of FIX version 4.1), rather that we are at version 4.1 _or higher_ (from the `>=`).

**Review by maxnis — APPROVED** [review]
Looks good

**Comment by gbirchmeier** [comment]
Oops-- earlier this week I fixed some of these myself (noticed the errors in my IDE), but I used InvariantCulture.  In retrospect, that was stupid of me, because Ordinal is clearly more appropriate.

Going to rebase your PR and fix those (newly created by me) conflicts.

**Pull request merged**